thunderbird-development/building-thunderbird/linux-build-prerequisites.md

```diff
@@ -4,41 +4,97 @@ description: >-
+You will need to be running a 64-bit version of Linux in order to build Thunderbird. You can check which version you're running by typing this command in your terminal:
 
 ```
 uname -m
 ```
 
 if this command returns `x86_64` you can proceed.
 
+## 30 GB of free space
+
+The Thunderbird build can use 30-40GB of disk space to complete depending on your operating system.
+
+{% hint style="warning" %}
+Note that while it's not technically required to have an internet connection to build, the default when building from `mozilla/comm-central` is that `--enable-bootstrap` is set so that the toolchains download automatically. If you do not have an active internet connection then 
+{% endhint %}
+
+# Build Environment
+
 ## Python
 
-You’ll need `Python 3.6` or later installed.
+You’ll need `Python 3.8` or later installed.
 
 You can check with `python3 --version` to see if you have it already. If not, you can install it with your distribution’s package manager. Make sure your system is up to date!
 
 You will also need `python3-distutils` and `python3-pip` installed from your distribution's package manager.
 
-## 30 GB of free space
+## Mercurial
+
+As noted in the [Getting Started page](../getting-started#mercurial-version-control), both `mozilla-central` and `comm-central` are version controlled with Mercurial. This means you will need to install Mercurial.
+Here are the quick commands to use for common Linux based operating systems but for a more complete list of instructions (if neither of these works for your use case), please see [Mercurial's download page on their wiki](https://www.mercurial-scm.org/wiki/Download).
+
+### Ubuntu/Debian
+```
+sudo apt install mercurial
+```
+
+### Fedora
+```
+sudo dnf install mercurial
+```
+
+# Getting the Code
+
+Once you have Mercurial installed, you are ready to grab the source code. There are a couple of different methods to do this.
 
-The Thunderbird build will take up to 30 GB of disk space in order to complete. Be sure to have enough free space and a fast internet connection to avoid interruptions.
+{% hint style="warning" %}
+Mozilla-central will build Firefox without the comm-central repo present and a few options set. Mozilla-central is the Firefox codebase and comm-central features the additions that turn Firefox into Thunderbird.
+{% endhint %}
+
+## Scripted
+
+We have created and host a script that will grab the two source repos you need, run `./mach bootstrap` for you, and sets up a necessary `mozconfig` file. This script is called [`bootstrap.py`](https://hg.mozilla.org/comm-central/raw-file/tip/python/rocboot/bin/bootstrap.py). Download this file to the directory where you would like your source code folder to live, either by clicking the link and moving the file to the appropriate location or using `wget`. Then we will make it executable and run it.
```

Comment:
Maybe this change?
We're switching from "(You) download this file and move it" to "We will make it...".

```suggestion
We have created and host a script that will grab the two source repos you need, run `./mach bootstrap` for you, and sets up a necessary `mozconfig` file. This script is called [`bootstrap.py`](https://hg.mozilla.org/comm-central/raw-file/tip/python/rocboot/bin/bootstrap.py). Download this file to the directory where you would like your source code folder to live, either by clicking the link and moving the file to the appropriate location or using `wget`. Then make it executable and run it.
```